modules/core/src/main/scala/com.snowplowanalytics/iglu.schemaddl/parquet/Migrations.scala

```diff
@@ -0,0 +1,223 @@
+package com.snowplowanalytics.iglu.schemaddl.parquet
+
+import com.snowplowanalytics.iglu.schemaddl.parquet.Type.{Array, Struct, DecimalPrecision}
+
+/*
+  Parquet schemas are migrated though the `merge.schema = true` loading option in Spark/Databricks.
+  Migrations in this package do not define spark 3.0 style `ATLER TABLE` transformations, similar to Redshift. Instead
+  they are used to categorize schema changes into 3 categories:
+   - Patch - Miscellaneous changes that would be handled by the `merge.schema`:
+      * `NullableRequired` - nullable to required change
+      * `NestedKeyAddition` - addition of the new Struct key anywhere except for top level Schema.
+      * `TypeWidening` - Compatible type casting. Such as Int -> Long (cast as BitInt) -> Decimal -> Double.
+   - Minor - Reserved for key addition in top level schema `TopLevelKeyAddition`.
+   - Major - Breaking change that would either lead to oen of these outcomes:
+      * `KeyRemoval` historical data loss, such as removing field in Struct. Loading will succeed with data loss.
+      * `IncompatibleType` Incompatible type casting, such as String -> Int, etc. Loading will fail.
+      * `RequiredNullable` Nullable to required field or array migration. Loading will fail.
```

Comment:
How did you decide on what defines a patch vs minor change?

These new rules are completely different to [the usual rules for schema versioning](https://snowplow.io/blog/introducing-schemaver-for-semantic-versioning-of-schemas/).  I'm not sure it's helpful to have competing versioning systems which look the same but are not.

A simpler implementation is to return simply `Breaking` or `NonBreaking` with respect to the loader -- what would we lose if we implemented it like that instead?